shade-test/jar-file/src/test/java/org/testcontainers/PublicAPITest.java

```diff
@@ -0,0 +1,120 @@
+package org.testcontainers;
+
+import lombok.RequiredArgsConstructor;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.FieldNode;
+import org.objectweb.asm.tree.MethodNode;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Parameterized.class)
+@RequiredArgsConstructor
+public class PublicAPITest extends AbstractJarFileTest {
```

Comment:
I've been reading this class 5 times for the last 15 minutes and I still can't figure it out. Maybe it's only me (more sleeping hours :)) but maybe we can refactor it a bit to make it more understandable:

* I don't understand the `@Parameters` mechanism. I mean, it's too complicated to grasp at first look. data() returns Object[] and not a specific type. You need to somehow understand that each array will be injected to a constructor for PublicAPITest which doesn't really exists, and is created by @RequiredArgsConstructor, which hand picks only the class members which are final. Then you need to know that all methods will be run for each list item. 
* I don't understand what data returns. Can we refactor the implementation inside to English look like code so when reading we will immediately understand what we get? 

I suggest:
* Replacing the usage of JUnit Parameterized to a simple test method, which looks like this:
```
    List<ClassItem> classesToCheck = getPubliclyAvaiableClasses()
    for (ClassItem classItem : classesToCheck) {
         verifySuperClass(classItem)
         verifyInterface(classItem);
        // ...
    }
```